core/src/main/java/org/apache/accumulo/core/client/IsolatedScanner.java

```diff
@@ -114,7 +114,8 @@ private void readRow() {
           }
 
           // wait a moment before retrying
-          sleepUninterruptibly(100, MILLISECONDS);
+          // ignore interrupt status
+          UtilWaitThread.sleep(100, MILLISECONDS);
```

Comment:
There are a lot of places in this PR where we are using this new method but intentionally ignoring the return value and the thread interrupt status. For example:
```
          // ignore interrupt status
          UtilWaitThread.sleep(100, MILLISECONDS);
```

I think this may be incorrect. If we are really going to ignore the thread interrupt status, then we should be resetting it back to `false` if it is `true`. I think, in the cases like this where we are ignoring the interrupt and return value, calling `Uninterruptibles.sleepUninterruptibly` is the right thing to do.

Reply:
Over time, we should look to improving the interrupt handling and this is a step in getting there.  Other than allowing an earlier return on interrupt, this preserves the current behavior.

The only difference between using `UtilWaitThread.sleep` and ignoring the interrupt as done here and using `sleepUninterruptibly` is that we will now exit the sleep on an interrupt (and log a message).  sleepUninterruptibly catches the interrupt and then goes back to sleep for the remaining time.  If an interrupt occurred, sleepUninterruptibly also sets the interrupted flag on return if an interrupt occurred, but we currently never look at it.

Reply:
> sleepUninterruptibly also sets the interrupted flag on return if an interrupt occurred

ok, I was lazy and didn't look. Thanks. Have you built this with `errorprone`? I'm wondering if it complains about the return value being unchecked. Is there a reason to return a `boolean` if the pattern is for the user to check if the Thread was interrupted after calling this method?

I would think that it would be more explicit to throw the InterruptedException when it's thrown in the current thread (e.g. Thread.sleep() ) and check the interrupt status on another thread.

Reply:
There are places where testing the return value seemed more explicit than testing the flag (see CompactionCoordinator, line 219)
```
 if (!UtilWaitThread.sleep(1000, MILLISECONDS)) {
        throw new InterruptedException("Interrupted during pause trying to get coordinator lock");
      }
```
In some places, this was used to throw an exiting exception (IOException, AccumuloException) 

I avoided throwing the InterruptedException because there are so many places where we currently do nothing and that seemed like we would need to add a lot of empty try / catches.  Not advocating that's the "best" way to handle things, but it is what we have now and this is trying to ease us into handling them, but that should be tacked over time.

Reply:
> there are so many places where we currently do nothing and that seemed like we would need to add a lot of empty try / catches

That's why I was thinking that continuing to use Uninterruptibles in these places makes sense. Ignore where you really want to ignore, handle where you really want to handle. I'm still curious what errorprone has to say.

Reply:
Running error prone explicitly (at the current level) says nothing.  I also was under the impression that we now run error prone by default, but I just did 
```
mvn verify -Perrorprone -Dtest=blah -Dit.test=blah
```
to double check.

Reply:
If we elect to continue to use Uninterruptibles I would like to hear the need that we *must* sleep for the entire time, ignoring the interrupt.

Running error prone at the next effort level, I do not see anything - is may be seeing that the interrupt flag is set as a side effect - but we fail pretty fast in core because of other (known) issues, so not really a strong test.

If it does get flagged, I would advocate that we create a similar method `UtilWaitThread.sleepQuitely` or something so that we still break out of the sleep on interrupts.